File: spring/src/test/java/com/example/PostControllerTest.java
package com.example;

import io.quarkus.test.InjectMock;
import io.quarkus.test.junit.QuarkusTest;
import io.restassured.http.ContentType;
import org.junit.jupiter.api.Test;
import org.springframework.data.domain.PageImpl;
import org.springframework.data.domain.Pageable;

import java.util.List;
import java.util.UUID;

import static io.restassured.RestAssured.given;
import static org.hamcrest.Matchers.equalTo;
import static org.mockito.ArgumentMatchers.isA;
import static org.mockito.Mockito.*;

@QuarkusTest
public class PostControllerTest {

    @InjectMock
    PostRepository postRepository;

    @Test
    public void testGetAll() {
        when(this.postRepository.findAll(isA(Pageable.class)))
                .thenReturn(
                        new PageImpl<Post>(
                                List.of(
                                        Post.of(UUID.randomUUID(), "foo", "bar"),
                                        Post.of(UUID.randomUUID(), "foo2", "bar2")
                                )
                        )
                );

        //@formatter:off
        given()
            .accept(ContentType.JSON)
        .when()
            .get("/posts")
        .then()
            .statusCode(200)
            .log().all()
            .body("content[0].title", equalTo("foo"));
        //@formatter:on

        verify(this.postRepository, times(1)).findAll(any(Pageable.class));
        verifyNoMoreInteractions(this.postRepository);
    }

}

File: restclient-kotlin-co/src/test/kotlin/com/example/demo/PostControllerTest.kt
package com.example.demo

import io.mockk.coEvery
import io.mockk.coVerify
import io.quarkiverse.test.junit.mockk.InjectMock
import io.quarkus.test.junit.QuarkusTest
import io.restassured.RestAssured.given
import io.restassured.http.ContentType
import kotlinx.coroutines.ExperimentalCoroutinesApi
import kotlinx.coroutines.test.runTest
import org.eclipse.microprofile.rest.client.inject.RestClient
import org.hamcrest.Matchers.equalTo
import org.junit.jupiter.api.Test
import java.time.LocalDateTime
import java.util.*

@QuarkusTest
class PostControllerTest {

    @InjectMock
    @field:RestClient
    lateinit var client: PostResourceClient

    @Test
    fun getPosts() = runTest {
        coEvery { client.getAllPosts(any(), any(), any()) } returns listOf(
            Post(id = UUID.randomUUID().toString(), title = "foo", content = "bar", createdAt = LocalDateTime.now()),
            Post(id = UUID.randomUUID().toString(), title = "foo2", content = "bar2", createdAt = LocalDateTime.now())
        )
        coEvery { client.countAllPosts(any()) } returns 15L

        //@formatter:off
        given()
            .accept(ContentType.JSON)
        .`when`()
            .get("/api")
        .then()
            .log().all()
            .statusCode(200)
            .body("count", equalTo(15))
        //@formatter:on

        coVerify(exactly = 1) { client.getAllPosts(any(), any(), any()) }
        coVerify(exactly = 1) { client.countAllPosts(any()) }
    }
}